pkg/manager/manager.go

```diff
@@ -131,5 +133,9 @@ func New(config *rest.Config, options Options) (Manager, error) {
 
 	cm.fieldIndexes = cm.cache
 	cm.client = client.DelegatingClient{ReadInterface: cm.cache, WriteInterface: writeObj}
+
+	cm.recorderProvider = recorder.NewProvider()
```

Comment:
Add a function to manager.go for calling the provider - e.g. add this to the Manager interface and implement it on manager.

```go
GetRecorder(name string) (record.EventRecorder, error)
```

Reply:
what the function for? you mean we should not take the provider as a field of manager, and pass it into the recorder injection?